### PR TITLE
cmd/thanos/receive: eliminate unused code

### PR DESCRIPTION
**Pull request by squat**
This commit simplifies the setup of the receive Handler. The
`reloadReady` variable was not actually doing anything and adds some
complexity. We can add this back in once reloading is actually
implemented.

## Changes
Remove some dead code in receive component setup.

cc @brancz @bwplotka @metalmatze

### CODE REVIEW

File: cmd/thanos/receive.go [hunk 1]
```diff
@@ -110,38 +109,13 @@ func runReceive(
...
-				case <-dbOpen:
-					break
-				case <-cancel:
-					reloadReady.Close()
-					return nil
-				}
-
-				reloadReady.Close()
-
+				<-dbOpen
```
Comment (brancz) [inline cmd/thanos/receive.go]: for this to not leak, we need to also close this channel in the tsdb run group no?

Comment (squat) [reply cmd/thanos/receive.go]: hmm now that I think about it, we could really eliminate this entire run group and put the single line:

```
webHandler.Ready()
```

in the tsdb run group, no?

Comment (squat) [reply cmd/thanos/receive.go]: in any case, we do already close it: https://github.com/improbable-eng/thanos/pull/1251/files#diff-074deab4cb04b87eb84800f5bbd8221fL176

Comment (brancz) [reply cmd/thanos/receive.go]: Not if we return right? https://github.com/improbable-eng/thanos/pull/1251/files#diff-074deab4cb04b87eb84800f5bbd8221fR144

Comment (squat) [reply cmd/thanos/receive.go]: ah yes, you're completely right. I'll add a defer

Comment (brancz) [reply cmd/thanos/receive.go]: But yes, if there is nothing else but the readiness thing, then let's just remove the group

Comment (squat) [reply cmd/thanos/receive.go]: ok simplified some more :) ptal when you get a change

**Comment by brancz** [comment]
Thanks!

**Pull request merged**